src/Categories/Multi/Category/Cartesian.agda

```diff
@@ -0,0 +1,115 @@
+{-# OPTIONS --without-K --safe #-}
+
+-- Idea:
+-- Whereas ordinary multicategories are based around the notion of substitution
+-- from noncommutative linear (Lambek) logic, Cartesian multicategories are
+-- based around the notion of substitution from intuitionistic logic, which is
+-- a lot simpler. A representable Cartesian multicategory is exactly a
+-- Cartesian category.
+--
+-- See https://ncatlab.org/nlab/show/cartesian+multicategory, particularly the
+-- section “Alternative presentation”
+-- (https://ncatlab.org/nlab/show/cartesian+multicategory#alternative_presentation).
+
+module Categories.Multi.Category.Cartesian where
+
+open import Data.List hiding ([_])
+open import Data.List.Membership.Propositional
+open import Data.Wrap
+open import Level
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+import Relation.Binary.Reasoning.Setoid as SetoidR
+
+infix 4 [_]_≈ᵉ_
+
+Env : ∀ {o ℓ} {X : Set o} (V : X → Set ℓ) (Δ : List X) → Set (o ⊔ ℓ)
+Env V Δ = ∀ {y} → y ∈ Δ → V y
+
+[_]_≈ᵉ_ : ∀ {o ℓ e} {X : Set o} {V : X → Set ℓ} {Δ} →
+  (∀ {x} → Rel (V x) e) → Rel (Env V Δ) (o ⊔ e)
+[_]_≈ᵉ_ = Wrap λ _≈_ ρ σ → ∀ {y} (i : y ∈ _) → ρ i ≈ σ i
+
+record CartesianMultiCategory (o ℓ e : Level) : Set (suc (o ⊔ ℓ ⊔ e)) where
+  infix 4 _≈_ _≈ˢ_ _⇒_ _⇒ˢ_
+  infixl 9 _∘_ _∘ˢ_
+
+  field
+    Obj : Set o
+    _⇒_ : List Obj → Obj → Set ℓ
+    _≈_ : ∀ {Γ A} (f g : Γ ⇒ A) → Set e
+
+  _⇒ˢ_ : List Obj → List Obj → Set (o ⊔ ℓ)
+  Γ ⇒ˢ Δ = Env (Γ ⇒_) Δ
+
+  _≈ˢ_ : ∀ {Γ Δ} (σ τ : Γ ⇒ˢ Δ) → Set (o ⊔ e)
+  _≈ˢ_ = [ _≈_ ]_≈ᵉ_
+
+  field
+    id : ∀ {Γ A} → A ∈ Γ → Γ ⇒ A
+    _∘_ : ∀ {Γ Δ A} → Δ ⇒ A → Γ ⇒ˢ Δ → Γ ⇒ A
+
+  idˢ : ∀ {Γ} → Γ ⇒ˢ Γ
+  idˢ = id
+
+  _∘ˢ_ : ∀ {Γ Δ Θ} → Δ ⇒ˢ Θ → Γ ⇒ˢ Δ → Γ ⇒ˢ Θ
+  (σ ∘ˢ τ) i = σ i ∘ τ
+
+  field
+    equiv : ∀ {Γ A} → IsEquivalence (_≈_ {Γ} {A})
+
+    ∘-resp-≈ : ∀ {Γ Δ A} {f f′ : Δ ⇒ A} {σ σ′ : Γ ⇒ˢ Δ} →
+      f ≈ f′ → σ ≈ˢ σ′ → f ∘ σ ≈ f′ ∘ σ′
+
+    identityˡ : ∀ {Γ Δ A} {i : A ∈ Δ} {σ : Γ ⇒ˢ Δ} → id i ∘ σ ≈ σ i
+    identityʳ : ∀ {Γ A} {f : Γ ⇒ A} → f ∘ idˢ ≈ f
+    identity² : ∀ {Γ A} {i : A ∈ Γ} → id i ∘ idˢ ≈ id i
+    assoc : ∀ {Γ Δ Θ A} {f : Θ ⇒ A} {σ : Δ ⇒ˢ Θ} {τ : Γ ⇒ˢ Δ} →
+      f ∘ σ ∘ τ ≈ f ∘ (σ ∘ˢ τ)
+    sym-assoc : ∀ {Γ Δ Θ A} {f : Θ ⇒ A} {σ : Δ ⇒ˢ Θ} {τ : Γ ⇒ˢ Δ} →
+      f ∘ (σ ∘ˢ τ) ≈ f ∘ σ ∘ τ
```

Comment:
Does `sym-assoc` achieve anything? It exists in `Category` so as to support `op` being a definitional involution. But there is no `op` for multicategories. There's an `op` for polycategories, but then I think the composition is so different that proofs would never just pass through.

Similar question for `identity²`. It at least feels like it still has a place, intermediate between the other two identity laws. But I can't tell whether any benefits based on that are worth the extra field.

Reply:
I would omit them until there's a compelling reason to add them. I'm sure some kinds of extra laws will arise when some things ought to be definitionally true - I just don't know what they are (yet).